pkg/codegen/go/gen_program.go

```diff
@@ -36,30 +37,22 @@ type generator struct {
 	arrayHelpers        map[string]*promptToInputArrayHelper
 	isErrAssigned       bool
 	configCreated       bool
+	createResourceVar   bool // Assign resource to a new variable instead of _.
 }
 
-func GenerateProgram(program *pcl.Program) (map[string][]byte, hcl.Diagnostics, error) {
-	// Linearize the nodes into an order appropriate for procedural code generation.
-	nodes := pcl.Linearize(program)
+// GenerateProgramOption functions are used to configure optional generator behavior.
+type GenerateProgramOption func(g *generator)
```

Comment:
Personally, I'm not a huge fan of functional options as it isn't great for discovering the options that can be passed-in. Wondering if a struct would be better.

```go
type GenerateProgramOptions struct {
	bool AssignResourcesToVariables // Assign resources to a new variable instead of _.
}

func GenerateProgram(program *pcl.Program) (map[string][]byte, hcl.Diagnostics, error) {
	// Generate program with default options.
	return GenerateProgramWithOptions(program, GenerateProgramOptions{})
}

func GenerateProgramWithOptions(program *pcl.Program, opts GenerateProgramOptions) (map[string][]byte, hcl.Diagnostics, error) {
	// copy over any "public" options from opts to the generator...
}
```

Reply:
Sure, that sounds good to me. I was initially thinking that I could modify GenerateProgram in a backward-compatible way, but that didn't work out, so there's not much point to keeping the functional opts.